base/src/main/java/proguard/evaluation/Variables.java

```diff
@@ -221,7 +232,12 @@ public Value load(int index)
      */
     public IntegerValue iload(int index)
     {
-        return load(index).integerValue();
+        try {
+            return load(index).integerValue();
+        } catch (InstructionEvaluationException e) {
```

Comment:
In what situation would `InstructionEvaluationException` get thrown? How can you be sure that it corresponds to the "Value in slot "+index+" is not an integer" message?

Reply:
A typical approach in ProGuard to make exceptions more informative is to catch earlier exceptions, chain them in a new exception, and add some local information in the message:

    catch (SomeException ex)
    {
        throw new SomeOtherException("More information ("+ex.getMessage()+")", ex);
    }

You may end up with a long message with nested parentheses, but at least you're not losing any information, even if the main code simply prints out that single message. In this case, you could catch IllegalArgumentException and NullPointerException, and add information about the variable index.

Reply:
I now added some commands regarding this. I think it is cleaner to not add the `StackInstructionEvaluationException` to the stack trace but if you disagree, I am willing to change this :)